crates/ruff_linter/src/rules/airflow/snapshots/ruff_linter__rules__airflow__tests__AIR302_AIR302_context.py.snap

```diff
@@ -0,0 +1,446 @@
+---
+source: crates/ruff_linter/src/rules/airflow/mod.rs
+snapshot_kind: text
+---
+AIR302_context.py:12:41: AIR302 `conf` is removed in Airflow 3.0
+   |
+11 | def access_invalid_key_in_context(**context):
+12 |     print("access invalid key", context["conf"])
+   |                                         ^^^^^^ AIR302
```

Comment:
This function doesn't have the `@task` decorator but the diagnostics is being raised here, is this correct?

Reply:
This is still correct as if the removed variables are part of `**context` in any airflow function, it should raise error.

Reply:
Can you briefly describe what is an "airflow function" or provide a reference to it?

Reply:
When I say function in airflow implementation. Let's say there is implementation of operator:
```
from airflow.models.baseoperator import BaseOperator

def log_execution_date(**context):
    # should throw an lint error
    execution_date = context.get("execution_date")
    if execution_date:
        print(f"Warning: 'execution_date' is deprecated. Found value: {execution_date}")
    else:
        print("No 'execution_date' found in context.")

class MyOperator(BaseOperator):
    def __init__(self, my_parameter, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.my_parameter = my_parameter

    def execute(self, context):
        log_execution_date(**context)
```

Reply:
Thanks for providing an example. So, in this case, the `log_execution_date` is an Airflow function, right?

Reply:
It's a user-defined function, but when `context` is passed through `BaseOperator.execute`, it should not access removed keys